.github/workflows/main.yml

```diff
@@ -0,0 +1,102 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+    - 'releases/**'
+  pull_request:
+    branches:
+    - main
+    - 'releases/**'
+  schedule:
+    - cron: "0 * * * *"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
The pull-request jobs need to be separate from the push jobs and move this registry login to only push jobs. 

> secrets are not passed to the runner when a workflow is triggered from a forked repository
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories